File: Template/binary_search.py

def find(arr, target):
    left = 0
    right = len(arr) - 1
    while left <= right:
        mid = int((left + right) / 2)
        if arr[mid] == target:
            return mid
        elif arr[mid] > target:
            right = mid - 1
        else:
            left = mid + 1
    return "not found"


def main():
    arr = [1, 3, 4, 5, 9, 10, 21, 33, 45, 100]
    print("find {} : {}".format(11, find(arr, 11)))
    print("find {} : {}".format(5, find(arr, 5)))
    return 0
    

if __name__ == "__main__":
    exit(main())

File: template_/BinarySearch.java

public class BinarySearch {

    public static int binarySearch(int[] array, int target) {
        int left = 0;
        int right = array.length - 1;
        while (left <= right) {
            int mid = (left + right) / 2;
            if (array[mid] == target) {
                return mid;
            } else if (array[mid] > target) {
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }
        return -1;

    }
    public static void main(String[] args) {
        int[] arr = {1, 3, 4, 5, 9, 10, 21, 33, 45, 100};       // 오름차순 정렬되어있는 상태
        System.out.println(BinarySearch.binarySearch(arr, 11));
        System.out.println(BinarySearch.binarySearch(arr, 5));
        // 존재하면 해당 인덱스 출력, 존재하지 않으면 -1 출력
    }
}
